Library/Formula/erlang.rb

```diff
@@ -39,11 +39,13 @@ class Erlang < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "libtool" => :build
-  depends_on "openssl"
   depends_on "unixodbc" if MacOS.version >= :mavericks
   depends_on "fop" => :optional # enables building PDF docs
   depends_on "wxmac" => :recommended # for GUI apps like observer
 
+  depends_on "libressl" => :optional
+  depends_on "openssl" unless build.with? "libressl"
```

Comment:
@DomT4 I've just tried it and it doesn't work this way. Namely, I've changed the same lines in `libevent.rb` from:

```
depends_on "libressl" => :optional
depends_on "openssl" unless build.with? "libressl"
```

to:

```
depends_on "openssl" => :recommended
depends_on "libressl" => :optional
```

Then I clean the cache at `brew --cache`.
Now when I try to install `libevent`, it installs `openssl`:

```
$ brew install libevent --with-libressl
==> Installing libevent dependency: openssl
==> Downloading https://homebrew.bintray.com/bottles/openssl-1.0.2.yosemite.bottle.tar.gz
^C
```

If I change it back and try to install it again, it works just fine:

```
$ brew install libevent --with-libressl
==> Downloading https://downloads.sourceforge.net/project/levent/libevent/libevent-2.0/libevent-2.0.2
^C
```